sklearn/utils/estimator_checks.py

```diff
@@ -1264,6 +1264,9 @@ def check_supervised_y_2d(name, estimator_orig):
 
 @ignore_warnings(category=(DeprecationWarning, FutureWarning))
 def check_classifiers_classes(name, classifier_orig):
+    # This is a pathological data set for ComplementNB.
```

Comment:
Can you elaborate on this one?
Can we improve the dataset?

Reply:
@TomDLT - the data set consists of three "[isotropic Gaussian blobs](http://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_blobs.html#sklearn.datasets.make_blobs)", which has no resemblance to text data, so the assumptions baked into CNB end up being detrimental. As a result, `ComplementNB` only predicts two out of three of the classes for this data set (it never predicts label "one"). I'm not sure how the data set could be changed to suit `ComplementNB` without also breaking other classifiers.
![blobs](https://user-images.githubusercontent.com/1927770/31551132-488e887c-aff9-11e7-9708-8f091eac24fc.png)

Reply:
But this should only exclude it from the `assert_array_equal(np.unique(y_), np.unique(y_pred))` assertion, not from `if np.any(classifier.classes_ != classes)` which should ideally still be checked.

Reply:
Fixed.